admin/class-yoast-form.php

```diff
@@ -30,6 +30,24 @@ class Yoast_Form {
 	 */
 	public $options;
 
+	/**
+	 * Option name for additional higher level options, if present.
+	 *
+	 * This can be the name of a set of network options in a multisite for example.
+	 *
+	 * @var string
+	 */
+	protected $override_option_name = '';
```

Comment:
I think this is not okay. This class should represent the form elements (in most ways it relies on option data). By adding this option logic it is getting an other dimension. Besides of that it is adding extra complexity. I think we should look for a better solution.

Reply:
Let's wait for @atimmer's review, and then decide how to proceed with this. Currently I introduce an abstraction in this PR that theoretically enables every site option to be disabled by a network option of the same name prefixed with `allow_`.

As this is currently only needed for the feature toggles though, we could also get rid of the abstraction and do it only for this very specific case. If we continue to pursue the abstracted approach, the `Yoast_Form` class needs to properly act upon it: If an option is disabled via its related `allow_...` network option being set to false, then the value for it cannot be updated (see the code in new `WPSEO_Option`). This needs to be reflected in the UI as well, therefore `Yoast_Form` would automatically need to render it as a disabled control.

If we decide to get rid of the abstraction and only handle this specific case of the feature toggle options, we could instead allow passing further arguments to each of the control output methods, so that the consuming code would be able to specify whether a control is disabled.